torch/csrc/Exceptions.h

```diff
@@ -373,15 +371,14 @@ using Arg = typename invoke_traits<Func>::template arg<i>::type;
 
 template <typename Func, size_t... Is>
 auto wrap_pybind_function_impl_(
+    // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
     Func&& f,
     std::index_sequence<Is...>,
     bool release_gil) {
-  using result_type = typename invoke_traits<Func>::result_type;
   namespace py = pybind11;
 
   // f=f is needed to handle function references on older compilers
-  return [f = std::forward<Func>(f),
-          release_gil](Arg<Func, Is>... args) -> result_type {
```

Comment:
Why was result_type removed from this lambda?

Reply:
I tried to reduce the dependency to C++17.h by using auto type deduction.

Reply:
Doesn't C++17  have any native APIs to do this? Presumably we can remove that dependency if we use the stdlib instead? Or does that cause issues on certain compilers?